### PR TITLE
migrate tree-sitter to conan2

### LINKED ISSUE #18547: [package] tree-sitter/*: tree-sitter needs to be updated to conan2.0
### Description

When I want to add `tree-sitter-cpp` lib recipe to CCI, the tree-sitter lib, which is required by tree-sitter-cpp, cannot be built because it doesn't support conan2.

### Expected
tree-sitter lib built successfully with conan2. 

### Package and Environment Details

* Package Name/Version: **tree-sitter/0.20.0**
* Operating System+version: **Arch Linux**
* Compiler+version: **GCC 13**
* Docker image: **conanio/gcc8**
* Conan version: **conan 2.0.4**
* Python version: **Python 3.9.12**

### Conan profile
```yaml
Host profile:
[settings]
arch=x86_64
build_type=Release
compiler=gcc
compiler.cppstd=gnu17
compiler.libcxx=libstdc++11
compiler.version=13.1
os=Linux

Build profile:
[settings]
arch=x86_64
build_type=Release
compiler=gcc
compiler.cppstd=gnu17
compiler.libcxx=libstdc++11
compiler.version=13.1
os=Linux
```

### Steps to reproduce
```shell
cd <CCI-repo>/recipes/tree-sitter
conan create all/conanfile.py --version 0.20.6 -pr:b=default -pr:h=default
```
### Logs
```shell
ERROR: Error loading conanfile at '/home/whalien/codebase/misc/conan-center-index/recipes/tree-sitter/all/conanfile.py': Unable to load conanfile in /home/whalien/codebase/misc/conan-center-index/recipes/tree-sitter/all/conanfile.py
  File "<frozen importlib._bootstrap_external>", line 940, in exec_module
  File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
  File "/home/whalien/codebase/misc/conan-center-index/recipes/tree-sitter/all/conanfile.py", line 1, in <module>
    from conans import CMake, ConanFile, tools
ImportError: cannot import name 'CMake' from 'conans' (/usr/lib/python3.11/site-packages/conans/__init__.py)
```

### PR DESCRIPTION
**Pull request by whaliendev**
this PR includes two revisions to tree-sitter lib in CCI:
+ migrated tree-sitter to conan2
+ bump version v0.20.8

I want to contribute tree-sitter-cpp grammar to CCI. However, it seems that the tree-sitter lib in CCI is broken due to the release of conan2. So I migrated it to conan2 and also follow upstream to release the latest version.

Closes #18547 

---

- [x] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [x] I've used a [recent](https://github.com/conan-io/conan/releases/latest) Conan client version close to the [currently deployed](https://github.com/conan-io/conan-center-index/blob/master/.c3i/config_v1.yml#L6).
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by RubenRBS** [comment]
Hi @whaliendev thanks a lot for your contribution, we really appreciate it :)

I've pushed a few minor changes to your branch, but nevertheless the changes look _great_!

**Comment by whaliendev** [comment]
@RubenRBS truly appreciate your help! Import all tools as the original author does so in v1 recipe, I'd like to preserve its work.

**Comment by whaliendev** [comment]
Any new progress on this PR? @RubenRBS 
How long will it take to get three approvals? 
As PR #18573 depends on this PR, how to speed up the integration?

**Comment by whaliendev** [comment]
> Any new progress on this PR? @RubenRBS How long will it take to get three approvals? As PR #18573 depends on this PR, how to speed up the integration?

PR #18702 also depends on this one now. 
@czoido  you can review this 3 recipes simultaneously. 
Also, for packages tree-sitter, tree-sitter-cpp, and tree-sitter-c, if you have any  homogeneous suggestions or improvement ideas, I can help modify all three of them. Once the modifications are done, I will ping you to merge these three. 
I'm not very familiar with conan recipe at the moment, so I truly appreciate your help.

**Comment by czoido** [comment]
> > Any new progress on this PR? @RubenRBS How long will it take to get three approvals? As PR #18573 depends on this PR, how to speed up the integration?
> 
> PR #18702 also depends on this one now. @czoido you can review this 3 recipes simultaneously. Also, for packages tree-sitter, tree-sitter-cpp, and tree-sitter-c, if you have any homogeneous suggestions or improvement ideas, I can help modify all three of them. Once the modifications are done, I will ping you to merge these three. I'm not very familiar with conan recipe at the moment, so I truly appreciate your help.

@whaliendev Thanks a lot for the contributions, I'll check the recipes and review, but It can take some days to merge as it also needs more reviewers and depends also on the load of the system.

File: recipes/tree-sitter/all/conanfile.py
Comment (czoido) [inline recipes/tree-sitter/all/conanfile.py:42]: This should be cmake_layout, I'll push the changes to fix that

**Comment by czoido** [comment]
Hi @whaliendev, I made some fixes to the recipe here: https://github.com/conan-io/conan-center-index/pull/18572/commits/c6ba7d70155aee56958915a829af37a66249a337 don't hesitate to ask about those if you have any questions. Let's wait for ci.

File: recipes/tree-sitter/all/conanfile.py
Comment (whaliendev) [reply recipes/tree-sitter/all/conanfile.py]: got it!

**Review by czoido — APPROVED** [review]

**Review by czoido — APPROVED** [review]

File: recipes/tree-sitter/all/conanfile.py
Comment (whaliendev) [reply recipes/tree-sitter/all/conanfile.py]: ok, now all looks good for me. tree-sitter-c should build successfully after this PR merged in. I'll self review the recipe later.

**Review by davidsanfal — APPROVED** [review]

**Pull request merged**
